src/com/ms/silverking/net/async/ConnectionStatsWriter.java

```diff
@@ -5,25 +5,25 @@
 import java.io.RandomAccessFile;
 
 public class ConnectionStatsWriter {
-    private final File   statsBaseDir;
-    
-    public ConnectionStatsWriter(File baseLogDir) {
-        statsBaseDir = baseLogDir;
-    }
-    
-    private File statsFile(Connection c) {
-        return new File(statsBaseDir, c.getRemoteSocketAddress().toString());
-    }
-    
-    public void writeStats(Connection c) throws IOException {
-        File                _statsFile;
-        RandomAccessFile    statsFile;
-        
-        _statsFile = statsFile(c);
-        _statsFile.getParentFile().mkdirs();
-        statsFile = new RandomAccessFile(_statsFile, "rw");
-        statsFile.seek(statsFile.length());
-        statsFile.writeBytes(System.currentTimeMillis() +"\t"+ c.statString() +"\n");
-        statsFile.close();
-    }
+  private final File statsBaseDir;
+
+  public ConnectionStatsWriter(File baseLogDir) {
+    statsBaseDir = baseLogDir;
+  }
+
+  private File statsFile(Connection c) {
+    return new File(statsBaseDir, c.getRemoteSocketAddress().toString());
+  }
+
+  public void writeStats(Connection c) throws IOException {
+    File _statsFile;
+    RandomAccessFile statsFile;
+
+    _statsFile = statsFile(c);
+    _statsFile.getParentFile().mkdirs();
+    statsFile = new RandomAccessFile(_statsFile, "rw");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *statsFile*. The resource is referenced in statements at the following lines: 25, 26. The resource closure statement is at line: 27. There are other execution paths that do not contain closure statements, for example, when RandomAccessFile.length throws an exception. Either a) close *statsFile* in a try-finally block or b) close the resource by declaring *statsFile* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).